tests/test_types.rs

```diff
@@ -73,6 +73,19 @@ fn test_vec() {
     assert_eq!(v, Ok(vec![1i32, 2, 3]));
 }
 
+#[test]
```

Comment:
This test actually passes on master, you need to wrap the result in a `Vec` to make it use from_redis_value**s**

```

#[test]
fn test_tuple() {
    use redis::{FromRedisValue, Value};

    let v = FromRedisValue::from_redis_value(&Value::Bulk(vec![Value::Bulk(vec![
        Value::Data("1".into()),
        Value::Data("2".into()),
        Value::Data("3".into()),
    ])]));

    assert_eq!(v, Ok(vec![(1i32, 2, 3,)]));
}

```

Reply:
I was able to assert _two_ levels of nested tuples using `let v` as you described: `assert_eq!(v, Ok(((1i32, 2, 3,),)));`

But this assertion failed:  `assert_eq!(v, Ok(vec![(1i32, 2, 3,)]));`

I'll check this against master branch too

Reply:
Ok, using your updated test case on master definitely fails, there:

```text
thread 'test_tuple' panicked at 'assertion failed: `(left == right)`
  left: `Err(Response was of incompatible type: "Bulk response of wrong dimension" (response was [bulk(string-data('"1"'), string-data('"2"'), string-data('"3"'))]))`,
 right: `Ok([(1, 2, 3)])`', tests/test_types.rs:86:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

So it seems like we're making progress :grin: